sway/tree/container.c

```diff
@@ -453,19 +453,25 @@ static void update_title_texture(struct sway_container *con,
 	int width = 0;
 	int height = con->title_height * scale;
 
-	cairo_t *c = cairo_create(NULL);
+	cairo_surface_t *fake_surface = cairo_image_surface_create(
+			CAIRO_FORMAT_ARGB32, 0, 0);
+	// somehow using NULL failed to fix #2869
```

Comment:
Is this a cairo bug?

Reply:
`cairo_create(NULL)` returns a nil:

> ```
> cairo_t *
> cairo_create (cairo_surface_t *target)
> {
>     if (unlikely (target == NULL))
> 	return _cairo_create_in_error (_cairo_error (CAIRO_STATUS_NULL_POINTER));
> ```

`cairo_set_font_options(nil)` is a no-op and `cairo_get_font_options(nil)` returns default font options.

> ```
> void
> cairo_set_font_options (cairo_t                    *cr,
> 			const cairo_font_options_t *options)
> {
>     cairo_status_t status;
> 
>     if (unlikely (cr->status))
> 	return;
> ```

> ```
> void
> cairo_get_font_options (cairo_t              *cr,
> 			cairo_font_options_t *options)
> {
>     /* check that we aren't trying to overwrite the nil object */
>     if (cairo_font_options_status (options))
> 	return;
> 
>     if (unlikely (cr->status)) {
> 	_cairo_font_options_init_default (options);
> 	return;
>     }
> ```

Therefore, I think such behavior is by-design. I'll update the (in-code) comment.

Reply:
Related: https://lists.cairographics.org/archives/cairo/2005-July/004448.html